lib/jormungandr/src/Cardano/Wallet/Jormungandr/Network.hs

```diff
@@ -456,28 +450,16 @@ withJormungandr
     -> (JormungandrConnParams -> IO a)
     -- ^ Action to run while node is running.
     -> IO (Either ErrStartup a)
-withJormungandr tr (JormungandrConfig stateDir block0 mPort logSeverity output extraArgs) cb =
-    bracket setupConfig cleanupConfig startBackend
-  where
-    nodeConfigFile = stateDir </> "jormungandr-config.yaml"
-    setupConfig = do
-        apiPort <- maybe getRandomPort pure mPort
-        p2pPort <- getRandomPort
-        let baseUrl = localhostBaseUrl $ fromIntegral apiPort
-        genConfigFile stateDir p2pPort baseUrl
-            & Yaml.encodeFile nodeConfigFile
-        logInfo tr $ mempty
-            <> "Generated Jörmungandr's configuration to: "
-            <> T.pack nodeConfigFile
-        pure (apiPort, baseUrl)
-    cleanupConfig _ = removeFile nodeConfigFile
-
-    startBackend (apiPort, baseUrl) = getGenesisBlockArg block0 >>= \case
+withJormungandr tr (JormungandrConfig stateDir block0 mPort logSeverity output extraArgs) cb = do
+    apiPort <- maybe getRandomPort pure mPort
+    let baseUrl = localhostBaseUrl $ fromIntegral apiPort
+    getGenesisBlockArg block0 >>= \case
         Right (block0H, genesisBlockArg) -> do
             let args = genesisBlockArg ++
-                    [ "--config", nodeConfigFile
+                    [ "--rest-listen", "127.0.0.1:" <> show apiPort
+                    , "--storage", stateDir </> "chain"
```

Comment:
Shouldn't `--storage` be left in the configuration file?

Reply:
I think we should really minimise the amount of control we put on Jormungandr's configuration unless we really have to, like the rest api listen.

Reply:
Can you clarify the use cases you have in mind? Because I do not believe that we need something infinitely flexible that works for every possible scenario. I think it's totally reasonable to set the storage option here.

Reply:
Well, that's my point actually. There's no use case for fixing the `storage` ourselves, same for the logging. They are already part of Jörmungandr configuration, so we can just leave them there and reduce our command-line surface. Which also means less testing for us. The only things we really care about is this `--rest-listen`, the rest, we can go with whatever value is defined in Jörmungandr's config file and don't need to do anything with them.

Reply:
Here are the three use cases I am addressing:
1. Daedalus runs `cardano-wallet-jormungandr launch`. They may or may not provide the `--state-dir` argument. They do not want to manage the chain storage directory and just need it to be in the same place as the other wallet backend files.
2. Typical CLI use case of `cardano-wallet-jormungandr launch` (e.g. exchange, or myself using the wallet server). User does not care where the chain data is, only that it's in the same place as the other wallet files. The ` --state-dir` option is usually omitted so that the default in `~/.local/share` is used.
3. Advanced use case - control of chain db directory is required. The user will run `cardano-wallet-jormungandr serve` and `jormungandr` themselves.

In cases 1 and 2, if the `--storage` option were not passed, then the user need to configure the state directory twice. Once for cardano-wallet and once for jormungandr. That's why I have put it there.